packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -41,4 +47,179 @@ void main() {
     await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
     expect(refreshCalled, true);
   });
+
+  testWidgets('RefreshIndicator - bottom', (WidgetTester tester) async {
+    refreshCalled = false;
+    await tester.pumpWidget(
+      new RefreshIndicator(
+        scrollableKey: scrollableKey,
+        refresh: refresh,
+        location: RefreshIndicatorLocation.bottom,
+        child: new Block(
+          scrollableKey: scrollableKey,
+          children: <String>['A'].map((String item) {
+            return new SizedBox(
+              height: 200.0,
+              child: new Text(item)
+            );
+          }).toList()
+        )
+      )
+    );
+
+    await tester.fling(find.text('A'), const Offset(0.0, -300.0), -1000.0);
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1)); // finish the scroll animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator settle animation
+    await tester.pump(const Duration(seconds: 1)); // finish the indicator hide animation
+    expect(refreshCalled, true);
+  });
+
+  testWidgets('RefreshIndicator - not enough', (WidgetTester tester) async {
+    refreshCalled = false;
+    await tester.pumpWidget(
+      new RefreshIndicator(
+        scrollableKey: scrollableKey,
+        refresh: refresh,
+        child: new Block(
+          scrollableKey: scrollableKey,
+          children: <String>['A'].map((String item) {
+            return new SizedBox(
+              height: 200.0,
+              child: new Text(item)
+            );
+          }).toList()
+        )
+      )
+    );
+
+    await tester.fling(find.text('A'), const Offset(0.0, 100.0), -1000.0);
```

Comment:
I'm not sure what this test is supposed to check. The title implies that that the fling hasn't been dragged far enough, but it looks like the fling is going in the wrong direction?